### DISCUSSION TITLE
Project Update: TinyDB Status 2024

### DISCUSSION THREAD

**Discussion by msiemens** [msiemens/tinydb#572] [General]
Dear Community,

Given the recent backlog of unanswered issues and PRs, I feel it's time to provide an update on the project's status and share what you can expect moving forward.

First off, I believe TinyDB has truly accomplished what it set out to do. It remains true to its name: a tiny database with a Pythonic interface and reasonable performance for small workloads. The project has now reached a staggering 6.8k stars on GitHub, which is absolutely mind-blowing to me! This clearly shows that the project has found its niche in the Python ecosystem.

Maintenance is still required, even for an 11-year-old project with thousands of daily downloads. Believe it or not, even after all this time, people still find unexpected corner cases in a 2,000-line Python program. But beyond fixing bugs, the increase in popularity naturally leads to an increase in questions, feedback and discussions.

Having worked at a medium-sized software development company for over two years, I've come to appreciate just how much work a good product manager can take off a development team's shoulders. It is *work* to sort through user feedback, coordinate all involved parties, and keep everyone up to date. While I'm fortunate to work with excellent PM colleagues professionally, for TinyDB (and my other OSS projects), I serve as the de facto product manager. Often, just responding to new comments can consume an entire evening, even without touching the code.

This leads to my final point: my involvement in OSS programming has significantly decreased. This shift began in 2021 when I started a new job with enough tough software engineering problems to solve during the day, leaving little desire to continue coding at night. Since then, there has been some OSS development on my GitHub profile, but it has been irregular (and in most cases work-related). Moreover, since late 2022 I've been dedicating my off-hours to exploring creative projects outside of programming. While I still care deeply about TinyDB, this change in focus has naturally affected the time and energy I can devote to open-source projects.

So, what does this mean for TinyDB going forward?

1. Maintenance mode: TinyDB will continue to exist and function, but expect a slower pace for updates and issue responses. I'll try to fix critical bugs as soon as possible, but I'm not planning any significant new features or architectural changes.
2. Community contributions: I'll continue to rely on the community to help with bug fixes and minor enhancements. If you're interested in contributing, please don't hesitate to submit PRs or reply to discussion posts.
3. Response times: I'll do my best to review issues and PRs periodically, but please be patient. It might take weeks for a response.
4. Release cadence: Releases will continue to be driven by user contributions and bug reports. I'll aim to bundle multiple changes into each release, typically waiting a few days (up to a week or two) after merging contributions to catch any additional updates.

I want to express my heartfelt gratitude to everyone who has used, supported, or contributed to TinyDB over the years. Your enthusiasm and feedback has been incredibly motivating and has gone (and continues to go!) above and beyond anything I could have asked for! While my active involvement may have decreased, I'm proud of what we've built together and the role TinyDB continues to play in the Python ecosystem.

Thanks, everyone!
Markus

**Comment by ShubhamKaudewar** [comment]
Thank you very much @msiemens for building such amazing innovative OSS TinyDB. I got to know about this today only after reading docs and implementing into my project. I realized this is an excellent project that really solved what I was looking for.

I keep forward to play more and integrate in future project. If possible I will try to contribute in any way. ♥️

**Comment by citrusvanilla** [comment]
Thanks for showing me the way brother. Cheers from Los Angeles
